Symfony/CS/Fixer/Contrib/ClassElementsOrderFixer.php

```diff
@@ -0,0 +1,305 @@
+     */
+    public function fix(\SplFileInfo $file, Tokens $tokens)
+    {
+        for ($i = 1, $count = $tokens->count(); $i < $count; ++$i) {
+            if (!$tokens[$i]->isClassy()) {
+                continue;
+            }
+
+            $i = $tokens->getNextTokenOfKind($i, array('{'));
+            $elements = $this->getElements($tokens, $i);
+
+            if (!$elements) {
+                continue;
+            }
+
+            $sorted = $this->sortElements($elements);
+            $endIndex = $elements[count($elements) - 1]['end'];
+
+            if ($sorted !== $elements) {
+                $this->sortTokens($tokens, $i, $endIndex, $sorted);
+            }
+
+            $i = $endIndex;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'Orders the elements of classes/interfaces/traits.';
+    }
+
+    private function findConfigurationPosition(array $types)
+    {
+        foreach ($types as $type) {
+            if (isset($this->config[$type])) {
+                return $this->config[$type];
+            }
+        }
+
+        foreach ($types as $type) {
+            $parents = self::$types[$type];
+
+            if ($parents && $pos = $this->findConfigurationPosition($parents)) {
+                return $pos;
+            }
+        }
+    }
+
+    private function getElements(Tokens $tokens, $startIndex)
+    {
+        ++$startIndex;
+        $elements = array();
+
+        while (true) {
+            $element = array(
+                'start' => $startIndex,
+                'visibility' => 'public',
+                'static' => false,
+            );
+            for ($i = $startIndex; ; ++$i) {
+                if ($tokens[$i]->equals('}')) {
+                    return $elements;
+                }
+
+                if ($tokens[$i]->isGivenKind(T_STATIC)) {
+                    $element['static'] = true;
+                    continue;
+                }
+                if ($tokens[$i]->isGivenKind(array(T_PROTECTED, T_PRIVATE))) {
+                    $element['visibility'] = strtolower($tokens[$i]->getContent());
+                    continue;
+                }
+
+                if ($tokens[$i]->isGivenKind(T_USE)) {
+                    $element['type'] = 'use_trait';
+                    $element['end'] = $this->findElementEnd($tokens, $i, array(';'));
+                    break;
+                }
+                if ($tokens[$i]->isGivenKind(T_CONST)) {
+                    $element['type'] = 'const';
+                    $element['end'] = $this->findElementEnd($tokens, $i, array(';'));
+                    break;
+                }
+                if ($tokens[$i]->isGivenKind(T_VARIABLE)) {
+                    $element['type'] = 'property';
+                    $element['end'] = $this->findElementEnd($tokens, $i, array(';'));
+                    break;
+                }
+                if ($tokens[$i]->isGivenKind(T_FUNCTION)) {
```

Comment:
It should work because closures are always within method bodies, and these are omitted by `findElementEnd`.
But we will see when I add necessary test cases. ;)